src/Git/Index/Update.hs

```diff
@@ -60,6 +69,41 @@ updateIndex relpath = do
       Normal ie -> Just ie
       _ -> Nothing
 
+parseContent :: MonadFail m => Parser a -> Path.AbsFile -> IO (m a)
+parseContent parser path = do
+    content <- openBinaryFile path ReadMode >>= LBS.hGetContents
+    pure $ lazyParseOnly (parser <* endOfInput) content
+
+getIgnores :: (MonadIO m, MonadError GitError m) => Path.AbsDir -> m Ignores
+getIgnores dirP = do
+    eMfIgnores <- liftIO $ (
+        try $ parseContent ignoresP $ dirP </> Path.relFile ".gitignore"
```

Comment:
I've been trying to to sprinkle special strings/file names around places. It's of course a lot easier with the static files like `.git` and the object store and stuff to hide their actual file paths. Do you see any benefit in abstracting this?

Reply:
I thought about it, but couldn't think of anywhere else that would need this, didn't do it on grounds of indirection. Can if you'd like.